wave_utils/include/wave/utils/measurement.hpp

```diff
@@ -0,0 +1,312 @@
+#ifndef WAVE_UTILS_MEASUREMENT_HPP
+#define WAVE_UTILS_MEASUREMENT_HPP
+
+#include <boost/multi_index_container.hpp>
+#include <boost/multi_index/member.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index/composite_key.hpp>
+#include <boost/version.hpp>
+#include <iterator>
+#include <chrono>
+
+namespace wave {
+
+using TimeType = std::chrono::steady_clock::time_point;
+
+/** Measurement type to store in the container */
+template<typename T, typename S>
+struct Measurement {
+    TimeType time_point;
+    S sensor_id;
+    T value;
+
+    Measurement(const TimeType &t, const S &s, const T &v) :
+            time_point{t},
+            sensor_id{s},
+            value{v} {}
+};
+
+/** Perform linear interpolation between two measurements
+ *
+ * This template can be specialized for measurement types requiring a
+ * different calculation
+ */
+template<typename T>
+decltype(T::value) interpolate(const T &m1, const T &m2, const TimeType &t) {
```

Comment:
decltype? What dees that do? Also for interpolate, should this indicate that it is purely linear interpolation?